### PR TITLE
CAPI Operator: move jobs to EKS prow cluster

### PR DESCRIPTION
**Pull request by furkatgofurov7**
Fixes: https://github.com/kubernetes-sigs/cluster-api-operator/issues/152
Part of: https://github.com/kubernetes-sigs/cluster-api-operator/issues/153
Related to: https://github.com/kubernetes/test-infra/issues/29722

/hold

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api-operator/cluster-api-operator-periodics-main.yaml [hunk 1]
```diff
@@ -14,12 +15,20 @@ periodics:
     - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20230513-7e1db2f1bb-1.25
       command:
       - "./scripts/ci-test.sh"
+      resources:
+        requests:
+          cpu: "1"
+          memory: "2Gi"
+        limits:
+          cpu: "1"
+          memory: "2Gi"
```
Comment (furkatgofurov7) [inline config/jobs/kubernetes-sigs/cluster-api-operator/cluster-api-operator-periodics-main.yaml:24]: @rjsadow @dims are these resource quotas standard and numbers fine? Do we need to increase anything, perhaps?

Comment (rjsadow) [reply config/jobs/kubernetes-sigs/cluster-api-operator/cluster-api-operator-periodics-main.yaml:24]: It's hard to say as each job is different. It's going to be best to work with the CAPI maintainers to find an initial set of values.

It'll be important that someone is watching the jobs for a while after this merges to see if jobs start failing from respurce backoff. I imagine most changes we have for this larger effort will go through some itterations.

Comment (furkatgofurov7) [reply config/jobs/kubernetes-sigs/cluster-api-operator/cluster-api-operator-periodics-main.yaml:24]: /hold 
until we have a generic resource quota numbers for the jobs that did not have them from the beginning

**Comment by Fedosin** [comment]
/hold

didn't notice https://github.com/kubernetes/test-infra/pull/29743#discussion_r1226399609

**Comment by ameukam** [comment]
@furkatgofurov7 Can you please rebase this PR ?

**Comment by furkatgofurov7** [comment]
> didn't notice [#29743 (comment)](https://github.com/kubernetes/test-infra/pull/29743#discussion_r1226399609)

I think we are good to go ahead with current resource quotas. I will be watching the jobs after this PR is merged and increase it if need arise.

/hold cancel

cc @Fedosin

**Comment by ameukam** [comment]
@furkatgofurov7 Please keep an eye of those jobs for a few days.

/approve
/lgtm

File: config/jobs/kubernetes-sigs/cluster-api-operator/cluster-api-operator-presubmits-main.yaml [hunk 2]
```diff
@@ -92,10 +125,18 @@ presubmits:
...
+            cpu: "1"
+            memory: "2Gi"
+          limits:
+            cpu: "1"
+            memory: "2Gi"
     annotations:
       testgrid-dashboards: sig-cluster-lifecycle-cluster-api-operator
       testgrid-tab-name: capi-operator-pr-test-main
   - name: pull-cluster-api-operator-e2e-main
+    cluster: eks-prow-build-cluster
```
Comment (furkatgofurov7) [inline config/jobs/kubernetes-sigs/cluster-api-operator/cluster-api-operator-presubmits-main.yaml:139]: @ameukam After the merge, this job started failing in https://github.com/kubernetes-sigs/cluster-api-operator/pull/143. Looking at the build logs in https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-operator/143/pull-cluster-api-operator-e2e-main/1668941993025212416 I don't find anything useful and just:

```
Job execution failed: Pod pending timeout.
Failed to open /logs/process-log.txt: open /logs/process-log.txt: no such file or directory
```

Comment (ameukam) [reply config/jobs/kubernetes-sigs/cluster-api-operator/cluster-api-operator-presubmits-main.yaml:139]: Looks like an issue with the build cluster. Let me ping folx on Slack.

Comment (furkatgofurov7) [reply config/jobs/kubernetes-sigs/cluster-api-operator/cluster-api-operator-presubmits-main.yaml:139]: Thanks, please tag me in the thread if possible

Comment (rjsadow) [reply config/jobs/kubernetes-sigs/cluster-api-operator/cluster-api-operator-presubmits-main.yaml:139]: +1 I'd love to follow along too. There have been a few merged jobs transitioning that have had issues so far.

**Pull request merged**